src/de/willuhn/jameica/hbci/TextSchluessel.java

```diff
@@ -123,16 +123,13 @@
       return list.toArray(new TextSchluessel[list.size()]);
 
     List<TextSchluessel> l = new ArrayList<TextSchluessel>();
-    for (int i=0;i<codes.length;++i)
+    for (String code : codes)
     {
-      for (int k=0;k<list.size();++k)
+      for (TextSchluessel ts : list)
       {
-        TextSchluessel ts = list.get(k);
-        String code = ts.getCode();
-        if (code.equals(codes[i]))
+        if (code.equals(ts.getCode()))
         {
           l.add(ts);
-          continue;
```

Comment:
Diese Aufruf von `continue` kann entfallen, da dies die letzte Anweisung in der Schleife ist und somit eh mit dem nächsten Element weitergemacht wird.

Reply:
Vermutlich war hier vorher eher ein `break` gemeint, da man nicht länger nach einem Textschlüssel für den i-ten Code suchen muss, wenn man bereits einen gefunden hat. Durch break geht es dann direkt beim (i+1)-ten Code weiter.

Reply:
Ja, vermutlich ist da ein break gemeint. Ist aber auch egal. Der ganze Code wird nur noch "passiv" im Archiv vom alten Inlandszahlungsverkehr benötigt. Der ist noch da, damit User noch die alten Zahlungsaufträge einsehen können.